osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/commands/security/GenerateAndReplaceKeyCommandExecutor.java

```diff
@@ -52,23 +54,23 @@ public GenerateAndReplaceKeyCommandExecutor() {
     @Override
     public ActionResponseDto executeBundleAction(final DlmsConnectionManager conn, final DlmsDevice device,
             final ActionRequestDto actionRequestDto) throws OsgpException {
-
-        return this.execute(conn, device, actionRequestDto);
+        return this.execute(conn, device, (CorrelatedObject<ActionRequestDto>) actionRequestDto);
     }
 
     @Override
     public ActionResponseDto execute(final DlmsConnectionManager conn, final DlmsDevice device,
-            final ActionRequestDto actionRequestDto) throws OsgpException {
+            final CorrelatedObject<ActionRequestDto> actionRequestDto) throws OsgpException {
```

Comment:
I must admit though that it probably is easier to read: CorrelatedObject vs CorrelatedActionRequestDto